### ISSUE TITLE
Add support for custom tags

### ISSUE THREAD

**Issue by chrismichaels84** [phoenix-elements/wysiwyg#1]
It would be nice to be able to easily add custom tags to the editor. For instance, if the user wanted to wrap a selection in `<span class="my-class"></span>` or `<pre><code></code></pre>`. As it stands, the user would have to manually add an if statement into the core js file, which breaks their repo from the upstream and causes forward compatibility issues.

My first thought is to add options at setup:
```javascript
$('#editor').wysiwyg({
  custom: {
    'name': ['<tag>', '</tag>'],
    'code': ['<pre><code>', '</code></pre>']
  }
});
```
And then use it as `<a data-edit="name">` which would wrap the current selection in that code.

**Comment by RandomlyKnighted** [comment]
Just a thought, but what about reading in the value of the data-edit attribute and then split it into an array. Naturally, if the array only has a length of 1 then you could compare it against an another array containing all the supported parameters for the execCommand function (such as bold, italic, etc). If it doesn't match anything in that array then you could just pass the full value to the execCommand like so:

```
document.execCommand("insertHTML", false, attribute + document.getSelection() + attribute);
```

where attribute is the value that was passed in from data-edit. This would not only allow for custom tags such as your examples above. Also, by doing it this way the developer using the editor would only have to use the data-edit attribute, which would simplify their JavaScript code since they wouldn't have to do add any options like you gave above.

**Comment by chrismichaels84** [comment]
What would that look like in the <a> tag for the toolbars?

**Comment by RandomlyKnighted** [comment]
What do you mean?

**Comment by chrismichaels84** [comment]
I don't understand how a developer would implement the idea when they
create their toolbar. What would the `<a data-edit="">` tag look like? Can
you give me an example? I like the idea of simplicity.

On Thursday, April 23, 2015, Tyler Hughes <notifications@github.com> wrote:
> What do you mean?
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci3.googleusercontent.com/proxy/GSmlrFKPP-EQoTk-Is8jOtjHQDYl3C6nPrGjKRycw4KiQ-ePaoGw5iniKyHZGoDSl3bgQWlfnMKDTV5FZ5QiRDuMbRoPCUXMFdZnIlSbUvBehsjwfmoTDC-iKvxPGY2n3RMHiWg47FUHRl-jE046HlLv9TNU4w=s0-d-e1-ft#https://github.com/notifications/beacon/AFKWPTHeSfdpAlIzSQNdMamZdUrhLQ42ks5oCQ2-gaJpZM4EGllX.gif
>

**Comment by RandomlyKnighted** [comment]
For example, when working with the Highlight.js library you have to wrap the code in ```<pre><code></code></pre>``` tags so by doing it this way the developer would just need to add the tag ```<a data-edit="pre code">``` and that would wrap it with both tags.

**Comment by chrismichaels84** [comment]
I like the idea of not having to add javascript code at setup, but I'm not sure that approach will work exactly. What about execCommands with attributes like `fontName Arial`. That would not have a length of one. Also, I'm not sure how often new execCommands are added, but there are dozens and I don't want to keep that list. 

What about something like
`<a data-edit="custom pre|code">`
That would allow allow for looking at the array's first index for the word "custom"

**Comment by RandomlyKnighted** [comment]
So if you're adding multiple tags you separate them with a vertical bar? I like that idea, but what happens if you want to include a class in the tag per your original example ```<span class="my-class"></span>```?

**Comment by chrismichaels84** [comment]
I think the best thing would be to give the user both options. Add it to
the setup or use simple pipes for tags.

**Comment by RandomlyKnighted** [comment]
I agree, I don't think we'd be able to do the class attribute without doing it in JavaScript setup options.

**Comment by chrismichaels84** [comment]
This all sounds good and fairly simple. If you want some experience, you're welcome to fork this repo and send me a pull request. Or I can tackle it on Monday.

**Comment by RandomlyKnighted** [comment]
Alright, I'll give it a try this weekend to see what I can come up with.
